ethereum_bridge/src/protocol/transactions/ethereum_events/events.rs

```diff
@@ -41,53 +46,76 @@ where
     D: 'static + DB + for<'iter> DBIter<'iter> + Sync,
     H: 'static + StorageHasher + Sync,
 {
+    let wrapped_native_erc20 = read_native_erc20_address(storage)?;
     let mut changed_keys = BTreeSet::default();
     for TransferToNamada {
         amount,
         asset,
         receiver,
     } in transfers
     {
-        let keys: wrapped_erc20s::Keys = asset.into();
-        let balance_key = keys.balance(receiver);
-        update::amount(storage, &balance_key, |balance| {
-            tracing::debug!(
-                %balance_key,
-                ?balance,
-                "Existing value found",
-            );
-            balance.receive(amount);
-            tracing::debug!(
-                %balance_key,
-                ?balance,
-                "New value calculated",
-            );
-        })?;
-        _ = changed_keys.insert(balance_key);
-
-        let supply_key = keys.supply();
-        update::amount(storage, &supply_key, |supply| {
-            tracing::debug!(
-                %supply_key,
-                ?supply,
-                "Existing value found",
-            );
-            supply.receive(amount);
-            tracing::debug!(
-                %supply_key,
-                ?supply,
-                "New value calculated",
-            );
-        })?;
-        _ = changed_keys.insert(supply_key);
+        if asset != &wrapped_native_erc20 {
+            let mut changed =
+                mint_wrapped_erc20s(storage, asset, receiver, amount)?;
+            changed_keys.append(&mut changed)
+        } else {
+            tracing::warn!(
+                "Redemption of the wrapped native token is not yet supported"
+            )
```

Comment:
maybe this is a bit silly, but we could log here the receiver and amount that would have been applied in storage

Reply:
Sure, have updated the e2e test to check those also